### ISSUE TITLE
Loot games dupe bug? or extra loot bug

### ISSUE THREAD

**Issue by rabitibike112** [GTNewHorizons/GT-New-Horizons-Modpack#17740]
### Your GTNH Discord Username

rabi.tiberius

### Your Pack Version

2.6.1

### Your Server

SP

### Java Version

Java 8

### Type of Server

Single Player

### Your Expectation

Loot games drops 4 chests of loot and no items randomly spawn in the middle of the room

### The Reality

![image](https://github.com/user-attachments/assets/fc410363-ed1b-4bef-9d1f-f92c1d19ae17)
Loot games dropped 4 chests of loot but also a lot of random loot in the middle of thje room, not in chests

### Your Proposal

No clue

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [ ] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [ ] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

**Comment by miaowwwwww** [comment]
how to reproduce? i recently played lootgame in 261but nothing happened

**Comment by rabitibike112** [comment]
That is the big problem, I have no clue what caused it and don't know how to replicate. I have played lootgames since and didn't repeat, it was a one-off

**Comment by miaowwwwww** [comment]
Well, with no way to reproduce, no one will be able to help

**Comment by rabitibike112** [comment]
Yeah, but by making the ticket others that had this happen to them might be able to help

**Comment by BegeistertsLab** [comment]
Since this is not reproducable and there has been no further information, I will go ahead and mark this as "stale" 

If it happens again, feel free to add another comment with further information and we will reinvestigate.

**Comment by Sinewhales** [comment]
Commenting because I experienced the same thing in 2.8.1 where the Loot Game generated rewards 7 times. No clue if it's reproducible or what could have caused this, I have opened a ton before and it's the first time it happened to me. 

<img width="1470" height="836" alt="Image" src="https://github.com/user-attachments/assets/ddcdbabb-865e-41bf-85b4-e3c6c28c11b4" />

**Comment by Worive** [comment]
Can confirm this occurred to me in Official Server on ETA in 2.7.#

**Comment by Worive** [comment]
#### TLDR

If the last action that cause to win reveals multiple neighbors tiles, it should trigger the bug. This can only occure at level 4, level 1, 2 and 3 are arealdy protected against this

Theoretically this is the reason why it's happening, but causing the bug explicitly is way more complex as it's RNG based about the design and how you reveal the game

#### Details

From looking at the code again, looks like it's caused because `revealField` calls `revealAllNeighbours` which then call again `revealField`, being recursive.

https://github.com/GTNewHorizons/LootGames/blob/master/src/main/java/ru/timeconqueror/lootgames/minigame/minesweeper/GameMineSweeper.java#L296-L330


And then we have `onLevelSuccessfullyFinished` which checks only if the lever is not less than 4, so when it gets triggered again and again, it still pass

https://github.com/GTNewHorizons/LootGames/blob/master/src/main/java/ru/timeconqueror/lootgames/minigame/minesweeper/GameMineSweeper.java#L110-L127
